### DISCUSSION TITLE
Welcome to Ansel Discussions!

### DISCUSSION THREAD

**Discussion by aurelienpierre** [aurelienpierreeng/ansel#21] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by aadm** [comment]
Hello AP & others!

Amateur photographer here, darktable user since 2018 on Linux, after being let down by Apple Aperture and Lightroom which made me face twice the difficult and stressfull transition of my photo archive from one system to another. I loved the idea of an open source solution that will not be shut down for commercial reasons and that will not lock down my archive. And when rdt was announced a few weeks ago I was glad to have a lean & mean version of my favourite application.

I appreciate the possibility to discuss here some of the issues that are not "proper" issues, such as the following: **lack of thumbnails captions in lighttable**. Why this is not a "proper" issue: because AP [states clearly in the wiki](https://github.com/aurelienpierre/R-Darktable/wiki) that these were removed for a reason:

> the extended overlays mode for thumbnails: each of the EXIF metadata display triggered its own SQL request per-thumbnail. LOSS OF CPU CYCLES. 

However, for the way I manage my library, having infos such as filename/version name, camera/lens/basic shooting params is fundamental for not being "lost" and understand without going to darkroom module differences between similar shots.

I also have a doubt about this, i.e. if this was really intended because perhaps what causes "loss of cpu cycles" are the "thumbnail tooltips" that are dynamically generated and show up as an additional textbox when hovering on an image, and not the "extended overlays" which seems to me are "fixed" and do not involve extra cpu work (but this is just an wild guess).

In short, what I'm asking is to take in consideration to bring back the extended overlays, which are those parameterized by the preference setting `preferences/lighttable/thumbnails/pattern for the thumbnail extended overlay text` (while I'm perfectly fine to remove altogether the volatile captions set in `preferences/lighttable/thumbnails/pattern for the thumbnail tooltip`).

**Reply by aurelienpierre** [reply]
> However, for the way I manage my library, having infos such as filename/version name, camera/lens/basic shooting params is fundamental for not being "lost" and understand without going to darkroom module differences between similar shots.

For this purpose, the module "Display metadata" has been made sticky in the bottom left corner of the lighttable. Its height is configurable through the usual (odd) `Ctrl+Scroll` and the metadata fields are configurable in the options (hamburger menu). The info will refresh depending on the hovered thumbnail.

> I also have a doubt about this, i.e. if this was really intended because perhaps what causes "loss of cpu cycles" are the "thumbnail tooltips" that are dynamically generated and show up as an additional textbox when hovering on an image, and not the "extended overlays" which seems to me are "fixed" and do not involve extra cpu work (but this is just an wild guess).

Tooltips have been removed too. The long story is there was one SQL query for the normal overlay info, another one for the extra line of extended overlays, and another one for the tooltips… although they showed all the same data.

**Reply by aadm** [reply]
> For this purpose, the module "Display metadata" has been made sticky in the bottom left corner of the lighttable. Its height is configurable through the usual (odd) `Ctrl+Scroll` and the metadata fields are configurable in the options (hamburger menu). The info will refresh depending on the hovered thumbnail.

Alright! I didn't realize it was possible to use the "Display metadata" and to customize it. I'm trying it and yes, it is slightly less immediate than having the caption just below the thumbnail but I can live with it.

**Comment by marcsitkin** [comment]
Hello AP and others

I'm a  retired advertising photographer and large format exhibition print
maker. I graduated RIT in 1974, many years before digital imaging became a
thing.

After the introduction of Photoshop in the mid 90's, it became my main
tool, and eventually Lightroom for 10 years. I started using darktable on a
Linux system in 2019, and have stuck with it ever since. I am no longer
pursuing commercial work, so my interests are different from a working pro.

I've read AP's statement, but still wonder who this fork is intended for.
While I agree on some of his statements, I don't really experience any
significant lag while processing images, and that is on a couple of 4 year
and older computers. I'm worried a bit that a couple of the modules I use
to style my photos are gone, and what I'll replace them with. While I
appreciate AP's concerns with correct color processing, sometimes I deviate
strongly from a "straight" interpretation, and need a forbidden module or
two. I hope that in the quest for lighter footprint and a faster
experience, we don't lose some of the choices we have in dt.

Any way, best of luck and I hope something great comes out of this.



On Sun, Jul 17, 2022 at 8:35 PM Aurélien PIERRE ***@***.***>
wrote:

> 👋 Welcome!
>
> We’re using Discussions as a place to connect with other members of our
> community. We hope that you:
>
>    - Ask questions you’re wondering about.
>    - Share ideas.
>    - Engage with other community members.
>    - Welcome others and are open-minded. Remember that this is a
>    community we
>    build together 💪.
>
> To get started, comment below with an introduction of yourself and tell us
> about what you do with this community.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aurelienpierre/R-Darktable/discussions/21>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN64KKUQ2AX4RNH5WSD6JNLVUSRFLANCNFSM532S6PUQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Marc Sitkin
50 images from Iceland 2021
<https://marcsitkin.smugmug.com/Iceland/Iceland-2021-50-Images/>
Marc Sitkin Photography Web Site <http://marcsitkin.smugmug.com>
Instagram <https://www.instagram.com/marcsitkin/>

**Reply by aurelienpierre** [reply]
> While I appreciate AP's concerns with correct color processing, sometimes I deviate
strongly from a "straight" interpretation, and need a forbidden module or
two. I hope that in the quest for lighter footprint and a faster
experience, we don't lose some of the choices we have in dt.

The remaining tools should give you the same features as the ones removed. Now, your ability to use them or not is a different topic, but that's easy to fix : https://photo.aurelienpierre.com/private-lessons-on-retouching-with-darktable/?lang=en

> I've read AP's statement, but still wonder who this fork is intended for.

Good question.

That would be photographers who want the best image quality possible with a streamlined editing experience that does not get in the way. Upstream dt is able to give you that image quality, but you will get a hay maze standing between you and the result : too many options actively harmful and explicitly non-recommanded but still there because there is this one dev that likes it, so keep away. Plus dozens of weird GUI quirks that don't align with any other previously-known mode of interaction with GUI (scrolling, selecting on hover to write, everything lowercase, core features accessible only from key shortcuts and not with a pointing device, etc.).

**Comment by lologor** [comment]
Salut Aurélien and others,

I'm an amateur photographer living in Germany for more than 30 years and still keeping my french ID. I'm photographing since more than 40 years now but never on a professional level. After switching to digital, I started to use a rather new and inexpensive software (90€?) named Lightroom. All went well until I bought a FujiFilm X-Trans sensor camera and Adobe changed their business model. Then I switched to Capture One. I'm a linux guy since 1994, so I always looked at the raw developping softwares there but I could not get my workflow adapted to them. This changed for me with Darktable 3.4. Also the moving to the scene refered approach is really a big step forward! So I through away Capture One and moved completly to Darktable (transition still not completly finished...). Going back to mystery boxes like Lightroom or Capture One is not an option any longer. 

On the UI side, the implementation is in total acceptable. I replace albums with tags and collections presets and yes it makes sense to filter the "album" then with some typical criteria like "grade in stars" or color tags. It is a good idea to seperate these two from my point of view. As a former programmer (pascal based languages... no C/C++) I'm a friend of lean and mean (and elegant) programming, yes in particular to keep maintenance and testing to an acceptable level. So I understand Aurélien your initiative but it makes me sad that Darktable has been forked that way. I was under the impression that Darktable had quite some strong architectural principals but still fattening happened. So basically I like your approach getting the fat out of the code, the UI, making things like print, maps etc much more optional (yes printing is a chapter for itself under Linux...).

My proposal is that strategy for the coming years of Darktable is discussed within the community - before code is proposed. I don't know how the scene refered approach has been decided in the past, but I see there a clear strategy to follow and module by module get it realised. I think that the community should decide on this strategy for the UI, for DAM etc before going into execution. This helps to formulate  clear release goals. This could help to "refuse" some developments that aren't not contributing to the strategy. I also hope that R&Darktable and Darktable wil one day remerge together again and that some lessons learned will help the programmers to come together again.

Until then I will try to make some time to make or discuss proposals together with the R&Darktable community. I realize that my message is quite long now. Sorry for that!

Cheers,
Laurent

**Reply by aadm** [reply]
> making things like print, maps etc much more optional (yes printing is a chapter for itself under Linux...).

Just a quick note, as I have noticed the concurrence of this particular comment and the issue raised by @aurelienpierre [about making some views optional](https://github.com/aurelienpierre/R-Darktable/issues/20): please do not scrap the print module! There are still people actually printing photographs at home like me! I have an old Canon Pro-100 and I enjoy the process of printing my photos, trying different paper types etc.

**Reply by lologor** [reply]
Fully clear, I also print at home, but since my Epson 3800 broke down, I replace it with a modern SC P900 and no chance to print under Linux. Maybe I still should consider the TurboPrint driver, that would mean to recalibrate all my papers (as it was the case with gutenprint)... So qemu is my friend in this case... I enjoy the same as you with your Canon Pro-100 (gutenprint driver?) but under MacOS with native Epson driver. Making a module like print "more" optional means for me, I think the same as Aurélien: Make it configurable to start with it or not.

**Reply by aadm** [reply]
Making the print module active on demand is totally fine for me, I just wanted to raise the issue of self-printing which is still a thing worth of consideration and as "photographers" (yes, amateur photographers in my case) we need to keep it alive.

I use turboprint, in its default settings works very well (I use Red River Paper and I only need to select glossy/matte paper like if it was Canon's paper types).

**Reply by aurelienpierre** [reply]
The goal is not to nuke the print view, but indeed to avoid initializing it if not enabled (even though my opinion is printing at home, for amateurs printing small volumes of pics, is a terribly expensive things – better use photo labs). The key idea here is that all and every Gtk widget that is declared in the UI gets initialized and computed, even if hidden from the user. Some of these Gtk widgets are custom-coded in darktable with disputable code, not really power-saving if you know what I mean. I fear that the people going all "I'm not bothered by the bloat, it's hidden in places I never go" don't quite realize that their CPU is very much bothered and still computes things.

**Reply by aurelienpierre** [reply]
> My proposal is that strategy for the coming years of Darktable is discussed within the community - before code is proposed. I don't know how the scene refered approach has been decided in the past, but I see there a clear strategy to follow and module by module get it realised.

The scene-referred approach has been the work of a single guy : me. I have built piece by piece a consistent set of tools meant to work together in a pipeline : filmic, tone equalizer, color balance, color calibration, blurs, diffuse & sharpen, the guided laplacian in highlights reconstruction. It was not a community project. There are no community projects anyway, contributions are anarchical. 

The community is a double-edged sword, because the darktable community is made mostly of IT guys turned hobbyists photographers who lack both practical and theoretical skills to make sensible decisions about what a photo editing software should be and how it should work.

What dt/R&Dt needs to improve is a small bunch of skilled photographers, pros or not, doing heavy-duty work really pushing the software on a regular basis. Amateurs doing lightweight work, especially if they are open-source evangelists, have too much patience towards GUI quirks and too low expectations in terms of visual results and workflow efficiency.

We also need more people trained and educated in color science, such that all the bug reports about visual stuff don't end up in people calling everything they don't understand or expect "artifacts". Being able to accurately communicate about color is a huge time saver and makes users able to diagnose subtle issues.

Basically, we need more ambition. Being just one of the 5 raw-processors for Linux (2% of desktop installs) and being happy about that is no way to improve. dt needs to be a serious picture editor software first, then an open-source soft. Right now, it's probably the best of FLOSS editors, which is not much.

**Reply by llajtamasi** [reply]
Ansel appears to be very good, but just thinking... why would I entrust my photography business to a platform developed and maintained by a single person - apparently fuelled by hubris and high technical ability? If that individual changes interest, redirects focus, gets hit by a bus (hope not) I (a photographer) am left with an unmaintainable white elephant with all my digital assets organized on its back. There is some level of redundancy in a chaotic open source community project contributed to by more than one - even if it reflects the old adage that a camel is really a horse designed by a committee. The other option (which I don't like - I have used and supported FOSS for many years) is to entrust my business assets to a proprietary  tool that I blindly pay for - but even they can go out of business or hold me to monthly ransom by switching to a subscription model.
...my 2 rupees worth

**Reply by aadm** [reply]
@llajtamasi I see your point but let's remember that Ansel is open source and it is at its core an opinionated, bloat-free fork of darktable. I'm pretty sure there are other coders/photographers out there with darktable programming experience that will take the burden of following up the work of Aurelien if he ever decides to stop working on this. After all, Ansel is pretty much robust and usable as it is now so worst case scenario it will be maintenance and keeping it up to date with supporting recent cameras etc. I am *not* expecting (nor I really want or need) anything silly like AI-driven sky-replacement and other absurdities that pop up in the what's new of commercial apps.

**Reply by mmoqui** [reply]
Not all AI-driven functionalities are absurdities in commercial raw editors. For example, the subject selection mask in LR is a great feature with which a time-consuming masking can be avoided with a lot of images to process. My goal isn't to enjoy the processing of images or to draw/set masks but to put intention in my images while processing them. The less time I'll spend on image processing, better I will be.

**Comment by marcsitkin** [comment]
I'm wondering why the display referred workflow modules are a choice. Is it for legacy reasons? Seems like a source of confusion.

**Reply by aurelienpierre** [reply]
What display-referred modules ? I have deprecated pretty much all of them. There is still the workflow option that I need to remove.

**Reply by marcsitkin** [reply]
The workflow option is what I was referring to. 
I have selected scene-referred workflow in preferences>processing
Should this choice still be there?

Choosing workflow:scene-referred>base
has a module still listed as display-referred:
Local Contrast


Choosing workflow:scene-referred>Color
has a module still listed as display-referred:
Color Zones

Choosing workflow:scene-referred>Correct
has a module still listed as display-referred:
Retouch


Choosing workflow:scene-referred>effect
has a few modules still listed as display-referred:
Watermark
Framing
Vignetting
Grain

My build is from last week (3.9.0+1708~g3db47c31e as listed in about), so you may have taken care of this.
I will do a new build this weekend.
Thanks

**Reply by aurelienpierre** [reply]
Yeah, give me some time. There are many details like that to update. I already removed more than 22 000 lines of code, can you believe what little functionality that represents ? #bloat

**Comment by marcsitkin** [comment]
I'm retired, take all the time you need. Hope it was helpful.

**Comment by bazmin** [comment]
Hi AP, I'm not a computer expert and very much an amateur photographer. The only reason I joined linux was because of darktable. I've felt that during the past couple of years dt has lost its way. I found that I was having to watch so many videos/read the manual to understand the latest changes - many of which just confused me. Processing photos should be fun. I echo your sentiments and a stripped down dt is just what I need! Compiling software is also beyond me so I'll follow this development. Good luck to you.

**Reply by aurelienpierre** [reply]
Hi ! Don't get your hopes too high though, processing photos is a job and requires some knowledge of colour theory, Ansel Adams zone system and other technical concerns that may make the learning curve not fun at all, whether the GUI has been simplified or not.

**Comment by olliwa-git** [comment]
Hello, amateur photographer since the 80s, I learned the basics of photography with a book that was recommended to me by a guy from the French cultural center in Amman, Jordan.  Mount Nebo, Petra, Jerash, Wadi Rum, were my first playgrounds!  With a so-called manual camera, 28 mm lens and kodachrome 25 ISO. Today, I do not meet the criteria mentioned in this discussion.  And I'm color blind! 
As I like to understand and master what I do, I will follow the development of this new branch with pleasure.  It reminds me the branch : blender for artists.  
For time being, I have compiled this branch, on debian amd64 wsl2 and ubuntu arm64 jetson nano,  even I'm windows user.  I like the idea of ​​activating modules as needed.  To print, the use of the export function allows printing with expert tools or sending them to a lab (I got very good results with online sites even with very large formats.). 
Glad to find again a thread about color science ...

**Comment by aadm** [comment]
Hello! Since @aurelienpierre is not on discuss.pixls.us anymore, and not having seen public activity on the repo, it's just difficult to understand the state of the project, if this is being taken care of by Aurelien, anyway I was just curious about it.

I should add that I have been using it exclusively for a while now -- it is stable and I can do everything that I want to do, but you know how spoiled we are with our hunger for new-new-new!

And if there's anything that simple users like us can help, let us know.

**Reply by gritty101** [reply]
Relax guys! He's still taking in money on Liberapay as "darktable developer" so is working quietly behind the scenes. Wait for the big X-mas surprise release of R&DT that leaves boring dt 4.2 in the dust!

**Reply by blankslatephoto** [reply]
That is wonderful to hear. It is their loss on pixils, but I have to admit, with work heating up for me, it has slipped in to check in.  

On an unrelated note, the install instructions on the readme for this "fork" has the rant about mainline dt, but then the rest is a copy paste of the main darktable info, including build instructions for mainline dt. 

I know for those of you on the cutting edge it is easy, but we might might have some more "followers" so to speak if we 

I would do it but I don't trust myself to write this up... there was a pixils thread that got me going, but I'm sometimes hesitant to reinstall as something unintended happens... 

(This is not a complaint to the group.  I've dumped snaps, rocking fedora, and am learning more... I believe more users from pixils would come, test, and co tribute with a little hand holding).

Rant over. Thank you all for what your are doing.

**Reply by blankslatephoto** [reply]
Sorry... written on phone :)

**Reply by aadm** [reply]
I didn't make any further comments here because basically we agree on all aspects -- I am also happy about the current state of r-dt and find it functional and don't miss anything critical (the only thing I would change is perhaps its name -- I don't like the ampersand "&"!!). I simply wanted to get a feeling for AP's committment and ideas for the future, and the users' usage and interest in this fork.

**Reply by blankslatephoto** [reply]
Some of my "concerns" seem to be "in progress" with the new "Ansel" thing Aurelien has posted... there should be install and build instructions there (eventually :)

Did I miss an annoucement here for that (am I blind)... if so, and if there is an easy path forward for building, I would have no problem being a tester :)

**Reply by blankslatephoto** [reply]
Just successfully "built" ansel from the master branch on Fedora 37.  Works great, and I am enjoying it (after having to update lensfun, which was a couple of steps, but seemed to require little "futzing" than under Ubuntu. 

Newbie question re:  If I were to write up the steps to adjust the README file, reflecting how to build stable and the master branch with the proper links (so new linux users can just do the "copy-paste" thing), what is the recommended way?   

A large portion of the linux community, especially new users, install from the terminal using a lot of "copy-paste".... making this easier might get more people involved at least as testers...

 Looking at "how to use git" online gives options that I don't think are either used today or are not "the way" people do this in 2023... hopefully I can learn this once, then help out with things that might not be top of mind to the group (like newbie instructions on how to get started/test). 

Lastly, and a little off topic: I'm waiting on some parts to show up for my pinebook pro and I'll try to build for ARM as well...I see the  "less of more" approach of Ansel working well for people who want to do light editing on the road with a 64 bit ARM setup.

**Reply by aurelienpierre** [reply]
> Hello! Since @aurelienpierre is not on discuss.pixls.us anymore, and not having seen public activity on the repo, it's just difficult to understand the state of the project, if this is being taken care of by Aurelien, anyway I was just curious about it.

Aurélien took the first real break from dt/Ansel since Covid hit. Business is back to normal. Happy new year !

**Comment by jbrogaard** [comment]
Very nice to see a new Win version - will Mac OS (installer) be supported in the future?

**Comment by mediatechnologic** [comment]
Started with DT 2 years ago parallel to LR. Before I was with C1 some years. Since one year I am completely with DT and left LR.
I fully understand and share the opinion of Aurelien about GeekTable 4.0.

Today I started with R-DT, I joined this GIT project, I joined LiberaPay and I started donations to Aurelien.
Not because I like monthly "fees", but because I like the quality of Aurelien.

I fully agree, that processing images requires knowledge (in color theory). But Aurelien, good knowledge is required not only for image processing, but for all "processing" areas (video, graphics, audio, cooking, art …) and everything, which has influence on the quite complicated and nonlinear human senses. Too many people think, that you need just a GUI or an App with a lot of options and you can process everything and you turn without any effort to an expert.

**Reply by gritty101** [reply]
Exactly! Who cares that the only thing he added in the last 3 months is a windows version. DT has had it for a long time. We don't need any more bloat. We should pay him to **not** add anything more! And it is great that we will now see a lot more windows users here. They know LR really well so have useful opinions!

**Comment by mediatechnologic** [comment]
Where can I find R-DT for windows? Building it from scratch does look quite complicated for someone who never did this ever.

**Reply by 520kaymann** [reply]
Windows version where oh where is it!?! Every time I compiled DT non forked fail as well

**Reply by olliwa-git** [reply]
fortunately for windows users, if I'm not wrong, "sa chum" uses a windows machine
:)

**Comment by olliwa-git** [comment]
wow, Ansel project looks very promising and very consistent
bravo Aurélien

**Comment by mmoqui** [comment]
Hello,

I'm an amateur photographer. I started in 2019 to develop my pictures from RAW. Because I'm a GNU/Linux user, commercial tools like Lightroom was not a possible solution and hence I looked for open-source solutions: Darktable and RawTherapee. After some tests, I chose Darktable for bad reasons; at the time I knew nothing about post-treatment and my tests were all about the feeling I had with the tools. But this was a good choice for the reason you, AP, were one of the developer who published videos and posts about having a good workflow for post-treatment with Darktable. By your work and mainly your explanations, I started to know what it means to develop an image from a RAW.

The reason I'm now interested by R&D and now Ansel is Darktable becomes, for me, a bloatware without any a clear direction. I'm now looking for a clean and simple solution in which all is built around our workflow (a scene-referred workflow) without any useless distractions. And for this I want a consistent, coherent and straightforward GUI.

My use of Darktable/Ansel is two:

-  as a RAW cataloger: I import my RAWs in order to keep track of them (by a full use of tags), to select those that can be in some of my current series, to edit regularly the images to keep them included or not into a series (and in their different editing steps), to mark with colors their step of development.
- as an image developer: I work only on RAWs that have succeeded the different rounds of editing. Even after some developments, I can revert the artistic choices previously made for new ones after some weeks or months of consideration.

For the final steps of editing, and mainly for the sequencing of my images, I use small prints (4"x6"). I rarely print from Darktable. The prints are made directly from SOOC JPEGs (I shoot both in JPEG and RAW). I eventually print the developed images for a final work on the sequence of the images. 

Currently, my difficulties in the development of images is to work on colors. For me, the use of Balance Color RGB isn't so straightforward for this. Nevertheless, beside the tool, I think I requires to gain some skills in this domain.

**Comment by dnlyko** [comment]
Greetings AP & fellow Ansel users.
I have been faithful to Nikon software since purchasing my first digital Nikon SLR.  Since DT was released for Windows, I have always dabbled in it but the frustration level was always unbearable.  I am very grateful for all the work AP has been doing.  I don't fully understand much of what AP says, but I am 100% convinced his goals are exactly inline with my convictions in what a RAW software solution should ultimately be.  I am proud to be onboard with Ansel.  I can not express in words how grateful I am that AP has created this fork.  The level of knowledge and dedication that AP has is exactly what is needed.

**Comment by aurelienpierre** [comment]
New discussions are moved on https://community.ansel.photos. Thanks !

**Reply by lu-k** [reply]
Since the hacking of the forum, discussions here have been reopened: https://github.com/aurelienpierreeng/ansel/discussions/725#discussion-9860822
